src/index.html

```diff
@@ -11,7 +11,168 @@
+            type="date"
+            name="dateOfBirth"
+            class="field"
+            placeholder="mm/dd/yyyy"
+          >
+        </label>
+
+        <label class="form-field">
+          I accept the term of the agreement
+          <input
+            type="checkbox"
+            name="acceptAgreement"
+            required
+          >
+        </label>
+      </fieldset>
+
+      <fieldset class="set">
+        <legend>Registration</legend>
+        <label class="form-field">
+          E-mail:
+          <input
+            type="email"
+            name="email"
+            class="field"
+            placeholder="email@example.com"
+            required
+          >
+        </label>
+
+        <label class="form-field">
+          Password:
+          <input
+            type="password"
+            name="password"
+            minlength="6"
+            maxlength="28"
+            required
+          >
+        </label>
+      </fieldset>
+
+      <fieldset class="set">
+        <legend>An interesting fact about you!</legend>
+        <div class="form-field field">
+          Do you love cats?
+          <label>
+            <input
+            type="radio"
+            name="loveCat"
+            value="Y"
+          >
+            Yes
+          </label>
+
+          <label>
+            <input
+            type="radio"
+            name="loveCat"
+            value="N"
+          >
+            No
+          </label>
+        </div>
+
+        <label class="form-field">
+          What is your favorite color?
+          <input
+          type="color"
+          name="color"
+          class="field"
+        >
+        </label>
+
+        <label class="form-field">
+          What time do you go to bed?
+          <input
+          type="time"
+          name="bedTime"
+          class="field"
+        >
+        </label>
+
+        <label class="form-field field">
+          What are your favorite brand of cars?
+          <select name="carsBrand" multiple>
+            <option value="bmw">BMW</option>
+            <option value="audi">Audi</option>
+            <option value="lada">Lada</option>
+          </select>
+        </label>
+
+        <label class="form-field">
+          How do you rate our work?
+          <input
+          type="range"
+          name="rate"
+          min="0"
+          max="100"
+          value="0"
+        >
+        </label>
+      </fieldset>
+
+      <fieldset class="set">
+        <legend>Additional info:</legend>
+        <label class="form-field field">
+          Comments:
+          <textarea name="comments"></textarea>
+        </label>
+
+        <label class="form-field">
+          Would you recommend us?
+          <select name="recommend">
+            <option value="y">yes</option>
+            <option value="n">no</option>
+          </select>
+        </label>
+
+      </fieldset>
+
+      <button type="submit">Submit</button>
+
+    </form>
+
   </body>
```

Comment:
```suggestion
      <button type="submit">Submit</button>
    </form>
  </body>
```
no need for blank lines between nested tags 😉